lesson_20/sources/MonsterDeck.cpp

```diff
@@ -0,0 +1,25 @@
+#include "MonsterDeck.h"
+
+class Monster;
+
+std::shared_ptr<Monster> MonsterDeck::generateMonster()
```

Comment:
Don't think someone else except MonsterDeck should handle Monster lifetime. Can return weak_ptr from shared_ptr, in this manner this code user will only have a way to validate Monster is still alive but won't extend its life under any circumstances

Reply:
done